# Finding Range,Variance,Standerd Deviation(first,second,third) With Different Datasets  -- Mesures of Dispersion / Spread

In [93]:
import pandas as pd
import numpy as np

In [2]:
diamonds=pd.read_csv(r"C:\Users\akash\Documents\seaborn_data\seaborn-data\diamonds.csv")
diamonds

carat        cut color clarity  depth  table  price     x     y     z
0       0.23      Ideal     E     SI2   61.5   55.0    326  3.95  3.98  2.43
1       0.21    Premium     E     SI1   59.8   61.0    326  3.89  3.84  2.31
2       0.23       Good     E     VS1   56.9   65.0    327  4.05  4.07  2.31
3       0.29    Premium     I     VS2   62.4   58.0    334  4.20  4.23  2.63
4       0.31       Good     J     SI2   63.3   58.0    335  4.34  4.35  2.75
...      ...        ...   ...     ...    ...    ...    ...   ...   ...   ...
53935   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75  5.76  3.50
53936   0.72       Good     D     SI1   63.1   55.0   2757  5.69  5.75  3.61
53937   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66  5.68  3.56
53938   0.86    Premium     H     SI2   61.0   58.0   2757  6.15  6.12  3.74
53939   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83  5.87  3.64

[53940 rows x 10 columns]

**Range**

In [3]:
depth=diamonds['depth']

In [11]:
Range=depth.max()-depth.min()

In [12]:
Range

36.0

**Variance**

In [13]:
variance=diamonds['depth'].var()

In [14]:
variance

2.05240384317702

**Standerd deviation**

In [4]:
depth_mean=diamonds['depth'].mean()

In [5]:
depth_std=depth.std()
depth_std

1.4326213188337733

In [6]:
first_std_upperBound = depth_mean + depth_std
first_std_upperBound 

63.18202621316081

In [9]:
first_std_lowerBound = depth_mean - depth_std
first_std_lowerBound 

60.31678357549326

* 1st upper bound standered deviation is 63.18202621316081
* 1st lower bound standered deviation is 60.31678357549326

In [12]:
second_std_upperbound = depth_mean + 2*depth_std
second_std_upperbound

64.61464753199458

In [13]:
second_std_lowerbound = depth_mean - 2*depth_std
second_std_lowerbound

58.884162256659486

* 2nd upper bound standered deviation is 64.61464753199458
* 2nd lower bound standered deviation is 58.884162256659486

In [37]:
third_std_upperbound = depth_mean + 3*depth_std
third_std_upperbound

66.04726885082836

In [38]:
third_std_lowerbound = depth_mean - 3*depth_std
third_std_lowerbound

57.45154093782571

* 3rd upper bound standered deviation is 66.04726885082836
* 3rd lower bound standered deviation is 57.45154093782571

In [25]:
taxis=pd.read_csv(r"C:\Users\akash\Documents\seaborn_data\seaborn-data\taxis.csv")
taxis

pickup              dropoff  passengers  distance  fare  \
0     2019-03-23 20:21:09  2019-03-23 20:27:24           1      1.60   7.0   
1     2019-03-04 16:11:55  2019-03-04 16:19:00           1      0.79   5.0   
2     2019-03-27 17:53:01  2019-03-27 18:00:25           1      1.37   7.5   
3     2019-03-10 01:23:59  2019-03-10 01:49:51           1      7.70  27.0   
4     2019-03-30 13:27:42  2019-03-30 13:37:14           3      2.16   9.0   
...                   ...                  ...         ...       ...   ...   
6428  2019-03-31 09:51:53  2019-03-31 09:55:27           1      0.75   4.5   
6429  2019-03-31 17:38:00  2019-03-31 18:34:23           1     18.74  58.0   
6430  2019-03-23 22:55:18  2019-03-23 23:14:25           1      4.14  16.0   
6431  2019-03-04 10:09:25  2019-03-04 10:14:29           1      1.12   6.0   
6432  2019-03-13 19:31:22  2019-03-13 19:48:02           1      3.85  15.0   

       tip  tolls  total   color      payment            pickup_zone  \
0     2.15    0.0  12.95  yellow  credit card        Lenox Hill West   
1     0.00    0.0   9.30  yellow         cash  Upper West Side South   
2     2.36    0.0  14.16  yellow  credit card          Alphabet City   
3     6.15    0.0  36.95  yellow  credit card              Hudson Sq   
4     1.10    0.0  13.40  yellow  credit card           Midtown East   
...    ...    ...    ...     ...          ...                    ...   
6428  1.06    0.0   6.36   green  credit card      East Harlem North   
6429  0.00    0.0  58.80   green  credit card                Jamaica   
6430  0.00    0.0  17.30   green         cash    Crown Heights North   
6431  0.00    0.0   6.80   green  credit card          East New York   
6432  3.36    0.0  20.16   green  credit card            Boerum Hill   

                          dropoff_zone pickup_borough dropoff_borough  
0                  UN/Turtle Bay South      Manhattan       Manhattan  
1                Upper West Side South      Manhattan       Manhattan  
2                         West Village      Manhattan       Manhattan  
3                       Yorkville West      Manhattan       Manhattan  
4                       Yorkville West      Manhattan       Manhattan  
...                                ...            ...             ...  
6428              Central Harlem North      Manhattan       Manhattan  
6429  East Concourse/Concourse Village         Queens           Bronx  
6430                    Bushwick North       Brooklyn        Brooklyn  
6431      East Flatbush/Remsen Village       Brooklyn        Brooklyn  
6432                   Windsor Terrace       Brooklyn        Brooklyn  

[6433 rows x 14 columns]

**Range**

In [27]:
tip=taxis['tip']

In [28]:
tip_range =tip.max() -tip.min()

In [29]:
tip_range

33.2

**Variance**

In [30]:
tip_variance =taxis['tip'].var()

In [31]:
tip_variance

5.9954440737928785

**Standard deviation**

In [32]:
tip_mean = taxis['tip'].mean()

In [33]:
tip_std = taxis['tip'].std()

In [34]:
first_std_upperBounce = tip_mean + tip_std
first_std_upperBounce

4.427779240322506

In [35]:
first_std_lowerBounce = tip_mean - tip_std
first_std_lowerBounce

-0.46933994294958503

* 1st upper bound standered deviation is 4.427779240322506
* 1st lower bound standered deviation is -0.46933994294958503

In [39]:
second_std_upperBounce = tip_mean + 2*tip_std
second_std_upperBounce

6.8763388319585514

In [40]:
second_std_lowerBounce = tip_mean - 2*tip_std
second_std_lowerBounce

-2.9178995345856302

* 2nd upper bound standered deviation is 6.8763388319585514
* 2nd lower bound standered deviation is -2.9178995345856302

In [41]:
third_std_upperBounce = tip_mean + 3*tip_std
third_std_upperBounce

9.324898423594597

In [42]:
third_std_lowerBounce = tip_mean - 3*tip_std
third_std_lowerBounce

-5.366459126221676

* 3rd upper bound standered deviation is 9.324898423594597
* 3rd lower bound standered deviation is -5.366459126221676

In [46]:
dots =pd.read_csv(r"C:\Users\akash\Documents\seaborn_data\seaborn-data\dots.csv")
dots

align choice  time  coherence  firing_rate
0    dots     T1   -80        0.0    33.189967
1    dots     T1   -80        3.2    31.691726
2    dots     T1   -80        6.4    34.279840
3    dots     T1   -80       12.8    32.631874
4    dots     T1   -80       25.6    35.060487
..    ...    ...   ...        ...          ...
843  sacc     T2   300        3.2    33.281734
844  sacc     T2   300        6.4    27.583979
845  sacc     T2   300       12.8    28.511530
846  sacc     T2   300       25.6    27.009804
847  sacc     T2   300       51.2    30.959302

[848 rows x 5 columns]

**Range**

In [47]:
coh = dots['coherence']

In [51]:
coh_range = coh.max() - coh.min()

In [52]:
coh_range

51.2

**Variance**

In [56]:
coh_variance =dots['coherence'].var()

In [57]:
coh_variance

238.81085830121913

**Standerd deviation**

In [58]:
coh_mean = dots['coherence'].mean()

In [59]:
coh_std = dots['coherence'].std()

In [60]:
first_std_upperbounce = coh_mean + coh_std
first_std_lowerbounce = coh_mean - coh_std
print(f"1st upper bound standered deviation is {first_std_upperbounce}")
print(f"1st lower bound standered deviation is {first_std_lowerbounce}")

1st upper bound standered deviation is 28.35161955080853
1st lower bound standered deviation is -2.555393135714194


In [61]:
second_std_upperbounce = coh_mean + 2*coh_std
second_std_lowerbounce = coh_mean - 2*coh_std
print(f"2nd upper bound standered deviation is {second_std_upperbounce}")
print(f"2nd lower bound standered deviation is {second_std_lowerbounce}")

2nd upper bound standered deviation is 43.80512589406989
2nd lower bound standered deviation is -18.008899478975557


In [62]:
third_std_upperbounce = coh_mean + 3*coh_std
third_std_lowerbounce = coh_mean - 3*coh_std
print(f"3rd upper bound standered deviation is {third_std_upperbounce}")
print(f"3rd lower bound standered deviation is {third_std_lowerbounce}")

3rd upper bound standered deviation is 59.258632237331256
3rd lower bound standered deviation is -33.46240582223692


In [63]:
exercise = pd.read_csv(r"C:\Users\akash\Documents\seaborn_data\seaborn-data\exercise.csv")
exercise

Unnamed: 0  id     diet  pulse    time     kind
0            0   1  low fat     85   1 min     rest
1            1   1  low fat     85  15 min     rest
2            2   1  low fat     88  30 min     rest
3            3   2  low fat     90   1 min     rest
4            4   2  low fat     92  15 min     rest
..         ...  ..      ...    ...     ...      ...
85          85  29   no fat    135  15 min  running
86          86  29   no fat    130  30 min  running
87          87  30   no fat     99   1 min  running
88          88  30   no fat    111  15 min  running
89          89  30   no fat    150  30 min  running

[90 rows x 6 columns]

**Range**

In [64]:
pulse = exercise['pulse']

In [65]:
pulse_range =pulse.max() - pulse.min()

In [66]:
pulse_range

70

**Variance**

In [70]:
pulse_variance =exercise['pulse'].var()

In [71]:
pulse_variance

220.7741573033708

**Standerd deviation**

In [69]:
pulse_mean = exercise['pulse'].mean()

In [73]:
pulse_std = exercise['pulse'].std()

In [75]:
first_std_upperbounce = pulse_mean + pulse_std
first_std_lowerbounce = pulse_mean - pulse_std
print(f"1st upper bound standerd deviation is {first_std_upperbounce}")
print(f"1st upper bound standerd deviation is {first_std_lowerbounce}")

1st upper bound standerd deviation is 114.55847089384943
1st upper bound standerd deviation is 84.84152910615057


In [76]:
second_std_upperbounce = pulse_mean + 2*pulse_std
second_std_lowerbounce = pulse_mean - 2*pulse_std
print(f"2nd upper bound standerd deviation is {second_std_upperbounce}")
print(f"2nd upper bound standerd deviation is {second_std_lowerbounce}")

2nd upper bound standerd deviation is 129.41694178769887
2nd upper bound standerd deviation is 69.98305821230113


In [77]:
third_std_upperbounce = pulse_mean + 3*pulse_std
third_std_lowerbounce = pulse_mean - 2*pulse_std
print(f"2nd upper bound standerd deviation is {second_std_upperbounce}")
print(f"2nd upper bound standerd deviation is {second_std_lowerbounce}")

2nd upper bound standerd deviation is 129.41694178769887
2nd upper bound standerd deviation is 69.98305821230113


In [78]:
attention = pd.read_csv(r"C:\Users\akash\Documents\seaborn_data\seaborn-data\attention.csv")
attention

Unnamed: 0  subject attention  solutions  score
0            0        1   divided          1    2.0
1            1        2   divided          1    3.0
2            2        3   divided          1    3.0
3            3        4   divided          1    5.0
4            4        5   divided          1    4.0
5            5        6   divided          1    5.0
6            6        7   divided          1    5.0
7            7        8   divided          1    5.0
8            8        9   divided          1    2.0
9            9       10   divided          1    6.0
10          10       11   focused          1    6.0
11          11       12   focused          1    8.0
12          12       13   focused          1    6.0
13          13       14   focused          1    8.0
14          14       15   focused          1    8.0
15          15       16   focused          1    6.0
16          16       17   focused          1    7.0
17          17       18   focused          1    7.0
18          18       19   focused          1    5.0
19          19       20   focused          1    6.0
20          20        1   divided          2    4.0
21          21        2   divided          2    4.0
22          22        3   divided          2    5.0
23          23        4   divided          2    7.0
24          24        5   divided          2    5.0
25          25        6   divided          2    5.0
26          26        7   divided          2    4.5
27          27        8   divided          2    7.0
28          28        9   divided          2    3.0
29          29       10   divided          2    5.0
30          30       11   focused          2    5.0
31          31       12   focused          2    9.0
32          32       13   focused          2    5.0
33          33       14   focused          2    8.0
34          34       15   focused          2    8.0
35          35       16   focused          2    8.0
36          36       17   focused          2    7.0
37          37       18   focused          2    8.0
38          38       19   focused          2    6.0
39          39       20   focused          2    6.0
40          40        1   divided          3    7.0
41          41        2   divided          3    5.0
42          42        3   divided          3    6.0
43          43        4   divided          3    5.0
44          44        5   divided          3    8.0
45          45        6   divided          3    6.0
46          46        7   divided          3    6.0
47          47        8   divided          3    8.0
48          48        9   divided          3    7.0
49          49       10   divided          3    6.0
50          50       11   focused          3    6.0
51          51       12   focused          3    8.0
52          52       13   focused          3    9.0
53          53       14   focused          3    7.0
54          54       15   focused          3    7.0
55          55       16   focused          3    7.0
56          56       17   focused          3    6.0
57          57       18   focused          3    6.0
58          58       19   focused          3    6.0
59          59       20   focused          3    5.0

**Range**

In [80]:
score= attention['score']

In [82]:
score_range = score.max() - score.min()
score_range

7.0

**Variance**

In [87]:
score_variance =attention['score'].var()

In [88]:
score_variance

2.629590395480226

**Standerd deviation**

In [84]:
score_mean = attention['score'].mean()

In [89]:
score_std = attention['score'].std()

In [90]:
first_std_upperbounce = score_mean + score_std
first_std_lowerbounce = score_mean - score_std
print(f"1st upper bound standerd deviation is {first_std_upperbounce}")
print(f"1st lower bound standerd deviation is {first_std_lowerbounce}")

1st upper bound standerd deviation is 7.579934515955073
1st lower bound standerd deviation is 4.336732150711593


In [91]:
second_std_upperbounce = score_mean + 2* score_std
second_std_lowerbounce = score_mean - 2* score_std
print(f"2nd upper bound standerd deviation is {second_std_upperbounce}")
print(f"2nd lower bound standerd deviation is {second_std_lowerbounce}")

2nd upper bound standerd deviation is 9.201535698576814
2nd lower bound standerd deviation is 2.7151309680898525


In [92]:
third_std_upperbounce = score_mean + 2* score_std
third_std_lowerbounce = score_mean - 2* score_std
print(f"3rd upper bound standerd deviation is {third_std_upperbounce}")
print(f"3rd lower bound standerd deviation is {third_std_lowerbounce}")

3rd upper bound standerd deviation is 9.201535698576814
3rd lower bound standerd deviation is 2.7151309680898525
